assets/src/edit-story/app/story/effects/test/useLoadStory.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { mount } from 'enzyme';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import useLoadStory from '../useLoadStory';
+import { createStory } from '../../../api/testUtils/story';
+import { useAPI, useHistory } from '../../../';
+
+jest.mock('../../../', () => ({
+  useHistory: jest.fn(),
+  useAPI: jest.fn(),
+}));
+
+function TestComponent({ restore }) {
+  useLoadStory({ storyId: 11, shouldLoad: true, restore });
+  return null;
+}
+
+describe('useLoadStory', () => {
+  let clearHistory;
+  let getStoryById;
+
+  beforeEach(() => {
+    useAPI.mockReset();
+    useHistory.mockReset();
+    clearHistory = jest.fn();
+    getStoryById = jest.fn();
+
+    useHistory.mockImplementation(() => ({ actions: { clearHistory } }));
+    useAPI.mockImplementation(() => ({ actions: { getStoryById } }));
+  });
+
+  it('should load publish date with correct utc timezone', async () => {
+    getStoryById.mockReturnValue(
+      Promise.resolve(
+        createStory({
+          date: '2020-01-01T19:20:20',
+          date_gmt: '2020-01-01T20:20:20',
+        })
+      )
+    );
+
+    const restore = jest.fn();
+    await mount(<TestComponent restore={restore} />);
+    expect(restore.mock.calls[0][0].story.date).toStrictEqual(
+      '2020-01-01T20:20:20Z'
+    );
```

Comment:
IMHO, this reads cleaner:

```js
    expect(restore).toHaveBeenCalledWith(
      expect.objectContaining(
        { story: { date: '2020-01-01T20:20:20Z' } },
      ),
    );
```

Reply:
Updated to use objectContaining.
objectContaining doesn't seem to work with nested levels, so I used it for the attributes under 'story', but not including 'story' itself.